### DISCUSSION TITLE
GPS communication over UART

### DISCUSSION THREAD

**Discussion by SkywalkerJLi** [bitcraze/discussions#815] [Q&A] [answered]
Hi!

I was wondering if the Crazyflie firmware still supports UART communication with a GPS module and if so if there is an example program that demonstrates this. After looking at the deprecated GpsTab, it looks like the module is simply adding variables to the LogConfig, but when I run the same code on my Crazyflie, I don't get any output.

Any sort of example code that demonstrates UART communication over the Rx Tx pins would be very helpful!

Thank you so much!

**Comment by ataffanel** [comment]
There is still some code in the Crayzflie that interface a GlobalTop gps: https://github.com/bitcraze/crazyflie-firmware/blob/e56b7fddf6e6ac9e10d9bacfffb4c4bfcd629219/src/deck/drivers/src/gtgps.c

This code is very old so I have no idea if it still works, though it does defines the log variable that are used by the GPS tab so it might be a good starting point.

**Reply by SkywalkerJLi** [reply]
Great, thank you so much!

I got the UART communication working and can receive GPS data from the module, however, the code frequently crashes the firmware of the crazyflie, and also takes a long time for the GPS module to get a fix.

I will continue trying to debug this issue and will let you know what I find out.

**Reply by ataffanel** [reply, marked as answer]
Good to hear! The long fix is a problem we have had in our experiments a long time ago. It was hard to attach a GPS module to the Crazyflie while keeping the antenna happy.

One though about the crashes: if you see weird assert in the RTOS (often in a list function, FreeRTOS is almost entirely made of lists ...), try to increase the task stack size. This is a very common problem. There is also a button to get  the tasks status and stack usage estimate in the client's log tab.

**Reply by SkywalkerJLi** [reply]
Great thank you so much!
